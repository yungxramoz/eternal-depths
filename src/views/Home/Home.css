.home-container {
    /* center content verticaly */
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

h1.main-title {
    text-align: center;
    font-size: 2rem;
    line-height: 2.5rem;
}

.button-menu {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-top: 2rem;
}

.button-menu button {
    width: 100%;
}