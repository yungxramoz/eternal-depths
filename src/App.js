import { Route, Routes } from 'react-router-dom'
import './App.css'
import Home from './views/Home/Home'
import NewCharacter from './views/NewCharacter/NewCharacter'

function App() {
  return (
    <div className="rpgui-content">
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/new-character" element={<NewCharacter />} />
      </Routes>
    </div>
  )
}

export default App
