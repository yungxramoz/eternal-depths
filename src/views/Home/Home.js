import { Link } from 'react-router-dom'
import RpgButton from '../../components/atoms/RpgButton/RpgButton'
import RpgContainer from '../../components/templates/RpgContainer/RpgContainer'
import RpgSeparator from '../../components/templates/RpgSeperator/RpgSeparator'
import './Home.css'

import bgImg from '../../assets/images/ui/home.png'

function Home() {
  return (
    <RpgContainer fullPage className="home-container" bgImg={bgImg}>
      <h1 className="main-title">Eternal Depths</h1>
      <RpgSeparator />
      <div className="button-menu">
        <Link to="/new-character">
          <RpgButton text="New Character" />
        </Link>
        <Link to="/leaderboard">
          <RpgButton text="Leaderboard" />
        </Link>
      </div>
    </RpgContainer>
  )
}

export default Home
