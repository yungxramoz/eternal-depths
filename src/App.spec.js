import React from 'react'
import TestRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import App from './App'
import Home from './views/Home/Home'

describe('App', () => {
  test('renders correctly', () => {
    const tree = TestRenderer.create(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders Home component for root path', async () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>,
    )
    const testInstance = testRenderer.root
    const homeComponents = await testInstance.findAllByType(Home)
    expect(homeComponents).toHaveLength(1)
  })
})
