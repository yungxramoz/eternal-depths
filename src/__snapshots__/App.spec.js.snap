// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly: App 1`] = `
<body>
  <div>
    <div
      class="rpgui-content"
    >
      <div
        class="rpgui-container container framed full-page scrollable home-container"
        style="background-image: url(home.png); background-repeat: no-repeat; background-size: cover; background-position: bottom;"
      >
        <h1
          class="main-title"
        >
          Eternal Depths
        </h1>
        <hr
          class="separator golden"
        />
        <div
          class="button-menu"
        >
          <a
            href="/new-character"
          >
            <button
              class="rpgui-button btn"
            >
              New Character
            </button>
          </a>
          <a
            href="/leaderboard"
          >
            <button
              class="rpgui-button btn"
            >
              Leaderboard
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
